src/mscorlib/src/System/Span.cs

```diff
@@ -625,5 +636,266 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
                 }
             }
         }
+
+        internal static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
+        {
+            if (byteLength == 0)
+                return;
+
+            fixed (byte* bytePointer = &b)
```

Comment:
> You should rewrite this using `ref byte`, without pinning. The pinning is unnecessary overhead - especially for small sizes.

So, removing pinning does improve perf for small sizes, but decreases perf for some larger sizes:
![image](https://cloud.githubusercontent.com/assets/6527137/23182289/e3273b34-f82c-11e6-96f2-f469648453fa.png)

I am guessing when you don't pin, the call to RhZeroMemory becomes more expensive:
```C#
            // P/Invoke into the native version for large lengths
            if (byteLength >= 512)
            {
                RuntimeImports.RhZeroMemory(ref b, byteLength);
                return;
            }
```